app/webpacker/controllers/multiple_checked_select_controller.js

```diff
@@ -0,0 +1,69 @@
+import { Controller } from "stimulus";
+
+export default class extends Controller {
+  static targets = ["button", "caret", "options", "option", "inputs", "filter"];
+  static values = { inputName: String };
+
+  connect() {
+    this.buttonTarget.addEventListener("click", this.toggleOptions);
+    this.optionTargets.forEach((option) => {
+      option.addEventListener("click", (e) =>
+        this.selectOption(e, option.dataset.value)
+      );
+    });
+    this.buildInputs();
+    document.addEventListener("click", this.closeOptions);
+    this.filterTarget.addEventListener("input", this.filterOptions);
+  }
+
+  disconnect() {
+    document.removeEventListener("click", this.closeOptions);
+  }
+
+  // private methods
+
+  filterOptions = (e) => {
+    const filter = e.target.value.toLowerCase();
+    this.optionTargets.forEach((option) => {
+      if (option.dataset["label"].toLowerCase().includes(filter)) {
+        option.classList.remove("hidden");
+      } else {
+        option.classList.add("hidden");
+      }
+    });
+  };
+
+  toggleOptions = () => {
+    this.optionsTarget.classList.toggle("hidden");
+    this.caretTarget.classList.toggle("icon-caret-down");
+    this.caretTarget.classList.toggle("icon-caret-up");
+  };
```

Comment:
Thinking of using more browser-native functionality. This is the functionality of a `<details>` element.

Reply:
I just discovered this `<details />` element:  https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details

I don't know exactly how it can be customized, and if it can fit into our specs (ie. having checkboxes into it)

Reply:
After reading a bit more, I don't think it's relevant for our use as it is : 
 - not relevant regarding the semantic
 -  _The `<summary>` element's contents can be any heading content, plain text, or HTML that can be used within a paragraph._: Which is not our case (contains inputs, divs, ...)

Reply:
Fair enough.